ext/openssl/ossl_ssl.c

```diff
@@ -2644,6 +2644,70 @@ ossl_ssl_tmp_key(VALUE self)
 	return Qnil;
     return ossl_pkey_new(key);
 }
+
+/*
+ * call-seq:
+ *    ssl.peer_signature_digest_name => String
+ *
+ * Returns the digest name string used by the peer to sign TLS messages.
+ */
+static VALUE
+ossl_ssl_get_peer_signature_digest_name(VALUE self)
+{
+    SSL *ssl;
+    const char *name;
+    int nid;
+
+    GetSSL(self, ssl);
+    if (!SSL_get_peer_signature_nid(ssl, &nid))
+        return Qnil;
+    if (nid == NID_undef)
+        return Qnil;
+    name = OBJ_nid2sn(nid);
+    if (!name)
+        return Qnil;
+    return rb_str_new2(name);
+}
+
+#ifdef HAVE_SSL_GET0_PEER_SIGNATURE_NAME
+/*
+ * call-seq:
+ *    ssl.peer_signature_name => String
+ *
+ * Returns the signature algorithms string used by the peer to sign the TLS
+ * handshake.
+ */
+static VALUE
+ossl_ssl_get_peer_signature_name(VALUE self)
+{
+    SSL *ssl;
+    const char *name;
+
+    GetSSL(self, ssl);
+    if (!SSL_get0_peer_signature_name(ssl, &name))
+        return Qnil;
+    return rb_str_new2(name);
```

Comment:
```suggestion
    return rb_str_new_cstr(name);
```

`new2` is a soft-deprecated macro for this function.